src/main/java/school/hei/haapi/repository/GroupRepository.java

```diff
@@ -1,9 +1,16 @@
 package school.hei.haapi.repository;
 
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 import school.hei.haapi.model.Group;
 
+import java.util.List;
+
 @Repository
 public interface GroupRepository extends JpaRepository<Group, String> {
+  @Query("select g from Group g")
+  Page<Group> getGroups(Pageable pageable);
```

Comment:
Why still getting Page<?> ? Don't you only want to have a Collection or a List ?